src/video_core/buffer_cache/buffer_cache.h

```diff
@@ -992,7 +992,25 @@ void BufferCache<P>::BindHostIndexBuffer() {
     TouchBuffer(buffer, index_buffer.buffer_id);
     const u32 offset = buffer.Offset(index_buffer.cpu_addr);
     const u32 size = index_buffer.size;
-    SynchronizeBuffer(buffer, index_buffer.cpu_addr, size);
+    if (maxwell3d->inline_index_draw_indexes.size()) {
+        if constexpr (USE_MEMORY_MAPS) {
+            boost::container::small_vector<BufferCopy, 4> copies;
+            auto upload_staging = runtime.UploadStagingBuffer(size);
+            copies.push_back(BufferCopy{
+                .src_offset = upload_staging.offset,
+                .dst_offset = 0,
+                .size = size,
+            });
+            const std::span<BufferCopy> copies_span(copies.data(), copies.size());
+            std::memcpy(upload_staging.mapped_span.data(),
+                        maxwell3d->inline_index_draw_indexes.data(), size);
+            runtime.CopyBuffer(buffer, upload_staging.buffer, copies_span);
```

Comment:
couldn't this just be
```suggestion
            auto upload_staging = runtime.UploadStagingBuffer(size);
            std::array<BufferCopy, 1> copies{{BufferCopy{
                .src_offset = upload_staging.offset,
                .dst_offset = 0,
                .size = size
            }}};
            std::memcpy(upload_staging.mapped_span.data(),
                        maxwell3d->inline_index_draw_indexes.data(), size);
            runtime.CopyBuffer(buffer, upload_staging.buffer, copies);
```